go/libkb/leveldb.go

```diff
@@ -158,7 +158,14 @@ func (l *LevelDb) doWhileOpenAndNukeIfCorrupted(action func() error) (err error)
 	// without resetting `dbOpenerOcce` (i.e. next call into LevelDb would result
 	// in a LevelDBOpenClosedError), because if DB open fails, retrying it
 	// wouldn't help. We should find the root cause and deal with it.
-
+	// MM: 10/12/2017: I am changing the above policy. I am not so sure retrying it won't help,
+	// we should at least try instead of auto returning LevelDBOpenClosederror.
+	if err != nil && l.db == nil {
+		l.Lock()
```

Comment:
Since we are checking on `l.db` above, it seems the lock should be held above the `if` above. Perhaps something like 
```
l.Lock()
defer l.Unlock()
if err != nil && l.db == nil {
 		l.G().Log.Debug("LevelDb: doWhileOpenAndNukeIfCorrupted: resetting sync one: %s", err)
 		l.dbOpenerOnce = new(sync.Once)
}
```